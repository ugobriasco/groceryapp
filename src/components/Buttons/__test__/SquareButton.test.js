import React from 'react';
import renderer from 'react-test-renderer';


import {SquareButton, styles} from '../';
import Icon from  'react-native-vector-icons/MaterialIcons';
import buildStyles from '../../../config/styles';


//do it if you care it
beforeAll(() => {
	buildStyles();
});

describe('SquareButton', () =>{

	it('renders correctly', () => {
	const tree = renderer.create(<SquareButton />)
	});

	it('styles an object', () => {
		expect(typeof styles).toBe('object');
	});

	it('should customize the button if icon name, color and size are set',()=>{
		const rendered = renderer.create(
			<SquareButton 
				iconName='add'
				size='40'
				color='blue'
				textColor='green'
				borderColor='grey'
			/>
		);
		expect(rendered).toMatchSnapshot();	
	});

	describe('props onPress', () => {
		it('should call a function', () => {
			const rendered = renderer.create(<SquareButton onPress={() => console.log('click')}/>);
			expect(rendered).toMatchSnapshot();	
		});
	});
});

const mockPressable = (name) => {
  const RealComponent = require.requireActual(name);

  class Component extends RealComponent {

    render() {
      return React.createElement(
        RealComponent.displayName || RealComponent.name,
        { ...this.props, onClick: this.props.onPress },
        this.props.children
      );
    }

  }

  return Component;
};


jest.mock('TouchableOpacity', () => mockPressable('TouchableOpacity'));