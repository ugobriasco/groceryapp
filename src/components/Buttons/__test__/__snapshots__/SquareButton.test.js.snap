// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SquareButton props onPress should call a function 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onClick={[Function]}
  onPress={[Function]}
  style={
    Array [
      Object {
        "borderRadius": 4,
        "borderWidth": 5,
      },
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#000",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 60,
          },
          Array [
            undefined,
            Object {
              "color": "#000",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</TouchableOpacity>
`;

exports[`SquareButton should customize the button if icon name, color and size are set 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  focusedOpacity={0.7}
  onClick={undefined}
  onPress={undefined}
  style={
    Array [
      Object {
        "borderRadius": 4,
        "borderWidth": 5,
      },
      Object {
        "backgroundColor": "blue",
        "borderColor": "grey",
      },
    ]
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 40,
          },
          Array [
            undefined,
            Object {
              "color": "green",
            },
          ],
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      î……
    </Text>
  </View>
</TouchableOpacity>
`;
